### PR TITLE
Update gemspec to use resolve mime-types 3.x for > ruby 2.x

### PR DESCRIPTION
**Pull request by milankubin**
using Mechanize 2.7.5 resolved ruby 2.x mime-types 3.x dependency.

### CODE REVIEW

**Comment by milankubin** [comment]
Test failed because ruby_dep 1.40 fails on requiring ruby >= 2.2.5 
see https://github.com/e2/ruby_dep/issues/20

Instead maybe we should bump requirement to ruby 2.3.1 stable?

**Comment by Marthyn** [comment]
Good suggestion. @commuun Would this be a problem for you?

File: lib/elmas/oauth.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ module Elmas
   module OAuth
     def authorize(user_name, password, options = {})
       agent = Mechanize.new
-
+      agent.log = Logger.new('log/mechanize.log')
```
Comment (Marthyn) [inline lib/elmas/oauth.rb]: Why is this log necessary? Does it not output to the standard logger?

**Comment by milankubin** [comment]
No it didn't! but it was also not my intention to leave it in the commit.
Apologies.
Should it normally?
This was purely for debugging authentication (which I still didn't get
working, might have something to do with SSL verifying, haven't tested that
yet).

**Comment by Marthyn** [comment]
@milankubin If you rebase on master the specs will also be fixed again.

Regarding your authentication problems, we realized that we didnt include the redirect uri in the readme. This config attribute is very important and has to be same as you set in your exact app. I revised the readme for that, that might be your issue with it.

**Comment by Marthyn** [comment]
@milankubin Is the authentication working for you now? i would like to proceed with this PR.

**Comment by karensg** [comment]
Would be great to update dependencies. The dependencies do not match with the modern gems any more..

**Comment by Marthyn** [comment]
@karensg Version 2.4.0 is out with updated gems

**Pull request closed without merging**